spec/wizards/placements/base_wizard_spec.rb

```diff
@@ -257,33 +247,40 @@
     subject { wizard.valid? }
 
     context "when all steps are valid" do
-      let(:session) do
+      let(:state) do
         {
-          "BurgerOrderWizard" => {
-            "choose_burger" => { "burger" => "beef" },
-            "meal_deal" => { "make_it_a_meal_deal" => "yes" },
-            "choose_side" => { "side" => "fries" },
-            "choose_drink" => { "drink" => "cola" },
-          },
+          "choose_burger" => { "burger" => "beef" },
+          "meal_deal" => { "make_it_a_meal_deal" => "yes" },
+          "choose_side" => { "side" => "fries" },
+          "choose_drink" => { "drink" => "cola" },
         }
       end
 
       it { is_expected.to be(true) }
     end
 
     context "when there's an invalid step" do
-      let(:session) do
+      let(:state) do
         {
-          "BurgerOrderWizard" => {
-            "choose_burger" => { "burger" => "beef" },
-            "meal_deal" => { "make_it_a_meal_deal" => "yes" },
-            "choose_side" => { "side" => "bread roll" }, # invalid
-            "choose_drink" => { "drink" => "cola" },
-          },
+          "choose_burger" => { "burger" => "beef" },
+          "meal_deal" => { "make_it_a_meal_deal" => "yes" },
+          "choose_side" => { "side" => "bread roll" }, # invalid
+          "choose_drink" => { "drink" => "cola" },
         }
       end
 
       it { is_expected.to be(false) }
     end
   end
+
+  describe ".generate_state_key" do
+    before do
+      allow(SecureRandom).to receive(:uuid).and_call_original
+      described_class.generate_state_key
+    end
+
+    it "returns a SecureRandom UUID" do
+      expect(SecureRandom).to have_received(:uuid).once
+    end
+  end
```

Comment:
Right now the test doesn't actually prove that we return the UUID – we only assert that it was called.

This might be a nicer way to test it and cover both the `SecureRandom.uuid` call and the expected return value.

```suggestion
  describe ".generate_state_key" do
    it "returns a SecureRandom UUID" do
      expect(SecureRandom).to receive(:uuid).once.and_return("pineapple")
      expect(described_class.generate_state_key).to eq("pineapple")
    end
  end
```